src/components/Text.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { forwardRef, Ref, RefObject, useEffect, useRef } from 'react';
+import {
+  Platform,
+  Text as RNText,
+  TextProps as RNTextProps,
+} from 'react-native';
+
+import { Gesture, GestureDetector } from '../';
+
+export const Text = forwardRef((props: RNTextProps, ref: Ref<RNText>) => {
+  const { children, onPress, ...rest } = props;
+
+  const textRef = useRef<RNText>(null);
+  const native = Gesture.Native().runOnJS(true);
+
+  useEffect(() => {
+    if (Platform.OS !== 'web') {
+      return;
+    }
+
+    const textElement = ref
+      ? (ref as RefObject<RNText>).current
+      : textRef.current;
+
+    // @ts-ignore at this point we are sure that text is div
+    textElement?.setAttribute('RNGHText', 'true');
+  }, []);
+
+  return (
+    <GestureDetector gesture={native}>
+      <RNText onPress={onPress} ref={ref ?? textRef} {...rest}>
```

Comment:
Can this ref be a function that would update both variables?

Reply:
Sure, done in [0ade4c0](https://github.com/software-mansion/react-native-gesture-handler/pull/3202/commits/0ade4c0aee5c8b0fb306a801c3587333d5137c99)